src/Bicep.Core/FileSystem/FileResolver.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.IO;
+using Bicep.Core.FileSystem;
+
+namespace Bicep.Core.FileSystem
+{
+    public class FileResolver : IFileResolver
+    {
+        public string GetNormalizedFileName(string filePath)
+            => PathHelper.ResolveAndNormalizePath(filePath);
+
+        public string? TryRead(string filePath, out string? failureMessage)
```

Comment:
Would returning a tuple make calling this easier? It's fine as-is - just curious.